cpp/CMakeLists.txt

```diff
@@ -124,3 +124,10 @@ add_executable(example_server examples/ws-server/src/main.cpp)
 target_compile_options(example_server PUBLIC ${SANITIZER_COMPILE_OPTIONS})
 target_link_options(example_server PUBLIC ${SANITIZER_LINK_OPTIONS})
 target_link_libraries(example_server foxglove_cpp_static)
+
+add_executable(example_mcap examples/mcap/src/main.cpp)
+target_compile_options(example_mcap PUBLIC ${SANITIZER_COMPILE_OPTIONS})
+target_link_options(example_mcap PUBLIC ${SANITIZER_LINK_OPTIONS})
+set_property(TARGET example_mcap PROPERTY CXX_STANDARD 17)
+set_property(TARGET example_mcap PROPERTY CXX_STANDARD_REQUIRED True)
```

Comment:
You can remove these two lines, I recently replaced them with a global cmake variable to fix some issues building the tests
```suggestion
```